org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -426,6 +426,9 @@ public synchronized void stop() {
 			return;
 		}
 		removeStopTimer();
+		if (this.languageClient != null) {
+            this.languageClient.dispose();
```

Comment:
In IJ usecase the LanguageClient implementation should add listener and we need a dispose to remove it.